docusaurus/docs/migration-guides/angular-react/suggestion-supplier.md

```diff
@@ -0,0 +1,78 @@
+---
+id: add-suggestion-supplier
+title: Add a suggestion supplier
+sidebar_position: 9
+description: How to add a suggestion supplier to a plugin.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - React
+  - ReactJS
+  - Angular
+  - migration
+  - suggestion
+---
+
+# Angular to React: Add a suggestion supplier
+
+You can add a suggestion supplier to examine query data coming from a panel and suggest usage of the plugin for the type of data detected. This guide provides instructions for doing so along with links to relevant examples.
+
+A single stat panel would be ranked "high" for a single series, and "low" for multiple series (or even none).
+
+## Add the suggestion supplier
+
+Here is an example suggestion suppler seen as part of `module.ts`:
+
+```ts
+import { MyDataSuggestionsSupplier } from './suggestions';
+...
+
+.setSuggestionsSupplier(new MyDataSuggestionsSupplier());
+```
+
+Here is an example suggestion supplier derived from polystat:
+
+```ts
+import { VisualizationSuggestionsBuilder } from '@grafana/data';
+import { MyOptions } from './types';
+
+export class MyDataSuggestionsSupplier {
+  getSuggestionsForData(builder: VisualizationSuggestionsBuilder) {
+    const { dataSummary: ds } = builder;
+
+    if (!ds.hasData) {
+      return;
+    }
+    if (!ds.hasNumberField) {
+      return;
+    }
+
+    const list = builder.getListAppender<MyOptions, {}>({
+      name: 'MyPanel',
+      pluginId: 'myorg-description-panel',
+      options: {},
+    });
+
+    list.append({
+      name: 'MyPanel',
+    });
+  }
+}
+```
+
+:::note
+
+When creating a suggestion supplier be certain the plugin can actually render something for the data being provided.
+
+:::
+
+## Additional resources
+
+Reference these suggestion suppliers to get ideas for further customization:
```

Comment:
Can you add a sentence or part of a sentence to note what's distinctive about the suggestion suppliers in each of these plugins?

Reply:
will do!